examples/tv-app/android/java/MessagesManager.cpp

```diff
@@ -0,0 +1,425 @@
+                                ChipLogError(Zcl, "Could not find MessageResponseOption constructor"));
+
+            auto iter = responses.Value().begin();
+            while (iter.Next())
+            {
+                auto & response = iter.GetValue();
+
+                jlong jniid = static_cast<jlong>(response.messageResponseID.Value());
+                UtfString jlabel(env, response.label.Value());
+
+                jobject jmessageResponseOption =
+                    env->NewObject(messageResponseOptionClass, messageResponseOptionCtor, jniid, jlabel.jniValue());
+
+                // add to vector
+                env->CallBooleanMethod(joptions, vectorAdd, jmessageResponseOption);
+                if (env->ExceptionCheck())
+                {
+                    ChipLogError(DeviceLayer, "Java exception in MessagesManager::HandlePresentMessagesRequest");
+                    env->ExceptionDescribe();
+                    env->ExceptionClear();
+                    return CHIP_ERROR_INTERNAL;
+                }
+            }
+        }
+
+        env->CallBooleanMethod(mMessagesManagerObject.ObjectRef(), mPresentMessagesMethod, jIdentifier.jniValue(), jpriority,
+                               jcontrol, jstartTime, jduration, jMessageText.jniValue(), joptions);
+        if (env->ExceptionCheck())
+        {
+            ChipLogError(DeviceLayer, "Java exception in MessagesManager::HandlePresentMessagesRequest");
+            env->ExceptionDescribe();
+            env->ExceptionClear();
+            return CHIP_ERROR_INTERNAL;
+        }
+    }
+    return CHIP_NO_ERROR;
+}
+
+CHIP_ERROR MessagesManager::HandleCancelMessagesRequest(const DataModel::DecodableList<ByteSpan> & messageIds)
+{
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+    VerifyOrReturnError(env != nullptr, CHIP_JNI_ERROR_NULL_OBJECT, ChipLogError(Zcl, "Could not get JNIEnv for current thread"));
+    JniLocalReferenceScope scope(env);
+
+    ChipLogProgress(Zcl, "Received MessagesManager::HandleCancelMessagesRequest");
+    VerifyOrReturnError(mMessagesManagerObject.HasValidObjectRef(), CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(Zcl, "Invalid mMessagesManagerObject"));
+    VerifyOrReturnError(mCancelMessagesMethod != nullptr, CHIP_ERROR_INCORRECT_STATE,
+                        ChipLogError(Zcl, "mCancelMessagesMethod null"));
+
+    env->ExceptionClear();
+
+    auto iter = messageIds.begin();
+    while (iter.Next())
+    {
+        auto & id = iter.GetValue();
+
+        char hex_buf[(kMessageIdLength * 2) + 1];
+        VerifyOrReturnError(CHIP_NO_ERROR ==
+                                chip::Encoding::BytesToUppercaseHexString(id.data(), id.size(), hex_buf, sizeof(hex_buf)),
+                            CHIP_ERROR_INCORRECT_STATE, ChipLogError(Zcl, "BytesToUppercaseHexString failed"));
+
+        UtfString jIdentifier(env, hex_buf);
```

Comment:
it seems UtfString is redundant, where UtfString create globalRef, and release it when it is out of scope in this function, it is better to use mEnv->NewStringUTF(hex_buf). when the data is passed from jni to java, Java would take care of the lifetime and decide when to release it.

Reply:
addressed